### ISSUE TITLE
GPT-OSS-120b agentic evals

### ISSUE THREAD

**Issue by ipastalTT** [tenstorrent/tt-inference-server#4903]
We need to record the reference scores on gpu for GPT-OSS-120b on agentic evaluation datasets.

## Plan

- [x] Terminal Bench 2.1
- ~~SWEBench-Verified~~
- [x] $t^3$-banking

**Comment by ipastalTT** [comment]
Ran into some issues with passing reasoning content to the server through harbor. 
Testing now on SC GPT-OSS and GPU in parallel.

Intermediate results (these were with medium effort most likely)




- $t^3$-banking

```json

{
  "id": "e22a7471-616e-473b-afd5-25782fdb1be0",
  "started_at": "2026-08-11T08:36:08.716970",
  "updated_at": "2026-08-11T09:27:16.117065",
  "finished_at": "2026-08-11T09:27:16.117065",
  "n_total_trials": 97,
  "stats": {
    "n_completed_trials": 97,
    "n_errored_trials": 3,
    "n_running_trials": 0,
    "n_pending_trials": 0,
    "n_cancelled_trials": 0,
    "n_retries": 0,
    "evals": {
      "tau3-llm-agent__openai/gpt-oss-120b__sierra-research/tau3-bench": {
        "n_trials": 97,
        "n_errors": 3,
        "metrics": [
          {
            "mean": 0.07216494845360824
          }
        ],
        "pass_at_k": {},
        "reward_stats": {
          "reward": {
            "0.0": [
              "tau3-bench__tau3-banking_knowled__eRrbZhL",
              "tau3-bench__tau3-banking_knowled__9SFKH9a",
              "tau3-bench__tau3-banking_knowled__zaTizv6",
              "tau3-bench__tau3-banking_knowled__tEhKyrb",
              "tau3-bench__tau3-banking_knowled__vvMmJgF",
              "tau3-bench__tau3-banking_knowled__P9Xzhz3",
              "tau3-bench__tau3-banking_knowled__UJopyYc",
              "tau3-bench__tau3-banking_knowled__FQEnWtH",
              "tau3-bench__tau3-banking_knowled__uL3ddcy",
              "tau3-bench__tau3-banking_knowled__5yZeyzP",
              "tau3-bench__tau3-banking_knowled__rB4mu9o",
              "tau3-bench__tau3-banking_knowled__pUNeANW",
              "tau3-bench__tau3-banking_knowled__cmv2F8n",
              "tau3-bench__tau3-banking_knowled__6TYe6nm",
              "tau3-bench__tau3-banking_knowled__z3rzBXo",
              "tau3-bench__tau3-banking_knowled__gNg3YWR",
              "tau3-bench__tau3-banking_knowled__DZAXQDc",
              "tau3-bench__tau3-banking_knowled__FV6a6Ld",
              "tau3-bench__tau3-banking_knowled__SdMTveD",
              "tau3-bench__tau3-banking_knowled__SbpZKrv",
              "tau3-bench__tau3-banking_knowled__t9bczfh",
              "tau3-bench__tau3-banking_knowled__aGSx77k",
              "tau3-bench__tau3-banking_knowled__GB7GC7g",
              "tau3-bench__tau3-banking_knowled__3RTp72G",
              "tau3-bench__tau3-banking_knowled__iBK9YtM",
              "tau3-bench__tau3-banking_knowled__KBMFA7e",
              "tau3-bench__tau3-banking_knowled__rw3DG9X",
              "tau3-bench__tau3-banking_knowled__n2hbA8B",
              "tau3-bench__tau3-banking_knowled__SRwL8zc",
              "tau3-bench__tau3-banking_knowled__ZRmYnfm",
              "tau3-bench__tau3-banking_knowled__E8HN3o4",
              "tau3-bench__tau3-banking_knowled__gtdWEpL",
              "tau3-bench__tau3-banking_knowled__YVqZ6SF",
              "tau3-bench__tau3-banking_knowled__bjcya43",
              "tau3-bench__tau3-banking_knowled__fcNabS2",
              "tau3-bench__tau3-banking_knowled__UdRFuhA",
              "tau3-bench__tau3-banking_knowled__WbmAS7G",
              "tau3-bench__tau3-banking_knowled__PK5NZpe",
              "tau3-bench__tau3-banking_knowled__bSe8CJc",
              "tau3-bench__tau3-banking_knowled__nYLA6U5",
              "tau3-bench__tau3-banking_knowled__dMjwaiz",
              "tau3-bench__tau3-banking_knowled__6JSMkPJ",
              "tau3-bench__tau3-banking_knowled__zMkhYRU",
              "tau3-bench__tau3-banking_knowled__RSkL9zR",
              "tau3-bench__tau3-banking_knowled__pSbk5YN",
              "tau3-bench__tau3-banking_knowled__EpoKCwn",
              "tau3-bench__tau3-banking_knowled__WFPt8ER",
              "tau3-bench__tau3-banking_knowled__uwVQpA9",
              "tau3-bench__tau3-banking_knowled__gFEsFYW",
              "tau3-bench__tau3-banking_knowled__NBk9CxC",
              "tau3-bench__tau3-banking_knowled__mgPk3ma",
              "tau3-bench__tau3-banking_knowled__WyBzmxE",
              "tau3-bench__tau3-banking_knowled__mccVXbD",
              "tau3-bench__tau3-banking_knowled__r63xVda",
              "tau3-bench__tau3-banking_knowled__6WjK6Ya",
              "tau3-bench__tau3-banking_knowled__t5TQ7Zn",
              "tau3-bench__tau3-banking_knowled__aVUZVpE",
              "tau3-bench__tau3-banking_knowled__CXxzxm7",
              "tau3-bench__tau3-banking_knowled__hRMfgen",
              "tau3-bench__tau3-banking_knowled__9TWytRW",
              "tau3-bench__tau3-banking_knowled__5Xy7udG",
              "tau3-bench__tau3-banking_knowled__DtUeKpx",
              "tau3-bench__tau3-banking_knowled__JbRvraK",
              "tau3-bench__tau3-banking_knowled__c5MSqzr",
              "tau3-bench__tau3-banking_knowled__Gy9BYKy",
              "tau3-bench__tau3-banking_knowled__eAgzrxC",
              "tau3-bench__tau3-banking_knowled__nSxx3eH",
              "tau3-bench__tau3-banking_knowled__hhRaicx",
              "tau3-bench__tau3-banking_knowled__DcARUwk",
              "tau3-bench__tau3-banking_knowled__vvkULfG",
              "tau3-bench__tau3-banking_knowled__XznJjPr",
              "tau3-bench__tau3-banking_knowled__9CUEZNs",
              "tau3-bench__tau3-banking_knowled__xBxTCs9",
              "tau3-bench__tau3-banking_knowled__8MuDnvt",
              "tau3-bench__tau3-banking_knowled__RJd4z92",
              "tau3-bench__tau3-banking_knowled__i8f34Zn",
              "tau3-bench__tau3-banking_knowled__6buf6DW",
              "tau3-bench__tau3-banking_knowled__VmSPgew",
              "tau3-bench__tau3-banking_knowled__UKZBcwY",
              "tau3-bench__tau3-banking_knowled__ZGUeLn2",
              "tau3-bench__tau3-banking_knowled__Vbv3zVF",
              "tau3-bench__tau3-banking_knowled__3PERkvg",
              "tau3-bench__tau3-banking_knowled__6Jseq4G",
              "tau3-bench__tau3-banking_knowled__AzXJCPN",
              "tau3-bench__tau3-banking_knowled__MLFHdMn",
              "tau3-bench__tau3-banking_knowled__9y8B6DG",
              "tau3-bench__tau3-banking_knowled__TAwiXjF",
              "tau3-bench__tau3-banking_knowled__L9kNJqz",
              "tau3-bench__tau3-banking_knowled__dDw7oTk",
              "tau3-bench__tau3-banking_knowled__QEego3o"
            ],
            "1.0": [
              "tau3-bench__tau3-banking_knowled__EvaBNd7",
              "tau3-bench__tau3-banking_knowled__3qqyfh7",
              "tau3-bench__tau3-banking_knowled__sXmn5ht",
              "tau3-bench__tau3-banking_knowled__YAbgNZb",
              "tau3-bench__tau3-banking_knowled__i5AnaBq",
              "tau3-bench__tau3-banking_knowled__xGdRSRY",
              "tau3-bench__tau3-banking_knowled__SzdK888"
            ]
          }
        },
        "exception_stats": {
          "NonZeroAgentExitCodeError": [
            "tau3-bench__tau3-banking_knowled__pUNeANW",
            "tau3-bench__tau3-banking_knowled__zMkhYRU",
            "tau3-bench__tau3-banking_knowled__xBxTCs9"
          ]
        }
      }
    },
    "n_input_tokens": 58790199,
    "n_cache_tokens": 0,
    "n_output_tokens": 722600,
    "cost_usd": 0.0
  },
  "_result_format": "harbor"
}
```

- terminal bench 2.1 (disk space got full at some point, hence the cancelled tasks)

```json

{
    "id": "3cd87e41-56b1-4c66-8109-4e63d2da3d99",
    "started_at": "2026-08-10T14:51:25.727004",
    "updated_at": "2026-08-10T15:37:10.862442Z",
    "finished_at": null,
    "n_total_trials": 89,
    "stats": {
        "n_completed_trials": 82,
        "n_errored_trials": 5,
        "n_running_trials": 7,
        "n_pending_trials": 0,
        "n_cancelled_trials": 4,
        "n_retries": 0,
        "evals": {
            "terminus-2__openai/gpt-oss-120b__terminal-bench/terminal-bench-2-1": {
                "n_trials": 77,
                "n_errors": 5,
                "metrics": [
                    {
                        "mean": 0.18292682926829268
                    }
                ],
                "pass_at_k": {},
                "reward_stats": {
                    "reward": {
                        "1.0": [
                            "kv-store-grpc__st9PyX5",
                            "openssl-selfsigned-cert__oKacrYc",
                            "pypi-server__dkhxrrH",
                            "cobol-modernization__RJcCvPk",
                            "merge-diff-arc-agi-task__S7Vy9HJ",
                            "pytorch-model-recovery__nNqE4vd",
                            "qemu-startup__BHi3aZt",
                            "overfull-hbox__Aoihtki",
                            "prove-plus-comm__M2Wz2x8",
                            "multi-source-data-merger__L6GnA27",
                            "fix-git__9SZnTLA",
                            "modernize-scientific-stack__sMVxyej",
                            "custom-memory-heap-crash__DZ4tz6M",
                            "constraints-scheduling__dezwuKS",
                            "build-pmars__TWNetrG"
                        ],
                        "0.0": [
                            "model-extraction-relu-logits__Jzt9SgW",
                            "regex-chess__mzke9wB",
                            "log-summary-date-ranges__MJdoZrv",
                            "mteb-leaderboard__d5SQnge",
                            "regex-log__YtPjbye",
                            "path-tracing__gz5qxpf",
                            "dna-assembly__oPQDSSP",
                            "gpt2-codegolf__gFdXN7L",
                            "schemelike-metacircular-eval__AYjZfp6",
                            "torch-tensor-parallelism__3Xssz2C",
                            "llm-inference-batching-scheduler__DzeVLoL",
                            "mteb-retrieve__rb9AUZ2",
                            "torch-pipeline-parallelism__wGd5pnM",
                            "nginx-request-logging__TPCVo6L",
                            "extract-elf__AjTW5SS",
                            "filter-js-from-html__weiomix",
                            "vulnerable-secret__kga6KwS",
                            "configure-git-webserver__LSxv6Qc",
                            "gcode-to-text__MYuKYCr",
                            "raman-fitting__Bxk5MAb",
                            "build-pov-ray__xWHboCk",
                            "adaptive-rejection-sampler__R4dAzQf",
                            "video-processing__RLqBiAC",
                            "sanitize-git-repo__mtFzJAS",
                            "feal-differential-cryptanalysis__tEb6MiS",
                            "reshard-c4-data__zDGFyej",
                            "dna-insert__DwVr4LR",
                            "make-doom-for-mips__j7YHtMS",
                            "hf-model-inference__nhe4N9G",
                            "sqlite-db-truncate__wyBuwgX",
                            "cancel-async-tasks__oQ3Ni2G",
                            "polyglot-c-py__HJYwtTx",
                            "financial-document-processor__ux76Equ",
                            "install-windows-3.11__BM5TuET",
                            "db-wal-recovery__8VCnk95",
                            "pytorch-model-cli__fgNQ7ev",
                            "circuit-fibsqrt__sFQEiXy",
                            "large-scale-text-editing__2Zs7QSD",
                            "count-dataset-tokens__wrzmGvG",
                            "query-optimize__G3PjWGY",
                            "path-tracing-reverse__nVHVQUP",
                            "extract-moves-from-video__ng49x2q",
                            "chess-best-move__EyqE3t8",
                            "break-filter-js-from-html__TnEUfG5",
                            "sam-cell-seg__GZgLcY7",
                            "git-multibranch__evkcfyi",
                            "polyglot-rust-c__UpaSsdH",
                            "feal-linear-cryptanalysis__k9fU2dS",
                            "sparql-university__4AP4iC9",
                            "password-recovery__MMFCDez",
                            "sqlite-with-gcov__VDxz4Zk",
                            "code-from-image__HsVzWmE",
                            "protein-assembly__aKJdreH",
                            "crack-7z-hash__WeRovgo",
                            "git-leak-recovery__hEHCtgB",
                            "distribution-search__RDPJnwp",
                            "bn-fit-modify__KAwz2af",
                            "largest-eigenval__EUaMKDq",
                            "fix-code-vulnerability__NZfQRe8",
                            "headless-terminal__4kXnNvR",
                            "winning-avg-corewars__cCkDSH5",
                            "fix-ocaml-gc__BjbS9Wn"
                        ]
                    }
                },
                "exception_stats": {
                    "RuntimeError": [
                        "caffe-cifar-10__obBxPjY"
                    ],
                    "CancelledError": [
                        "train-fasttext__Fa8beeE",
                        "build-cython-ext__rCPAtpv",
                        "mcmc-sampling-stan__kZiwAz6",
                        "compile-compcert__saoS6HH"
                    ]
                }
            }
        },
        "n_input_tokens": 14820834,
        "n_cache_tokens": 0,
        "n_output_tokens": 787561,
        "cost_usd": null
    }
}
```

**Comment by ipastalTT** [comment]
Found gpu cycles for this. Updated results:

## Tenstorrent Model Release Summary: gpt-oss-120b on SUPER_CLUSTER

### Metadata: gpt-oss-120b on SUPER_CLUSTER

```json
{
    "model_name": "gpt-oss-120b",
    "device": "SUPER_CLUSTER",
    "generated_at": "2026-08-23 20:23:48",
    "report_id": "id_gpt-oss_gpt-oss-120b_super_cluster_2026-08-23_20-23-48",
    "workflow": "agentic",
    "server_mode": "API",
    "run_command": "python run.py --model gpt-oss-120b --workflow agentic --device super_cluster --server-url 172.20.0.1:8000 --skip-system-sw-validation --dev-mode",
    "runtime_model_spec_json": "/home/ipastaltzidis/tt-inference-server/workflow_logs/runtime_model_specs/runtime_model_spec_2026-08-23_16-43-42_id_gpt-oss_gpt-oss-120b_super_cluster_dUXQ6SpS.json",
    "model_id": "id_gpt-oss_gpt-oss-120b_super_cluster",
    "model_repo": "openai/gpt-oss-120b",
    "inference_engine": "media",
    "tt_metal_commit": null,
    "vllm_commit": null,
    "model_impl": "gpt-oss"
}
```

### Acceptance Criteria

- Acceptance status: ✅ `PASS`
- Model status: `EXPERIMENTAL`
- Benchmarks: 🟨 `NA` (no blocks present)
- Evals: ✅ `PASS` (2/2 passed)
- Spec Tests: 🟨 `NA` (no blocks present)
- All acceptance criteria passed.

---

### Accuracy Evaluations for gpt-oss-120b on SUPER_CLUSTER

| Task               | Tolerance | Published Score | Published Score Ref                                                              | GPU Reference Score | Score | Ratio to Published | Ratio to Reference | Accuracy Check | mean_seconds_per_task |
|:-------------------|:----------|:----------------|:---------------------------------------------------------------------------------|:--------------------|:------|:-------------------|:-------------------|:---------------|:----------------------|
| terminal_bench_2_1 | 0.05      | 26.2            | https://artificialanalysis.ai/evaluations/terminalbench-v2-1?models=gpt-oss-120b | 26.2                | 33.71 | 1.287              | 1.287              | ✅ PASS        | 148.3                 |
| tau3_bench_banking | 0.1       | 12.8            | https://artificialanalysis.ai/evaluations/tau3-banking?models=gpt-oss-120b       | 7.2                 | 8.247 | 0.6443             | 1.145              | ✅ PASS        | 290.3                 |

Note: The ratio to published scores defines if eval ran roughly correctly, as the exact methodology of the model publisher cannot always be reproduced. For this reason the accuracy check is based first on being equivalent to the GPU reference within a +/- tolerance. If a value GPU reference is not available, the accuracy check is based on the direct ratio to the published score.
